Examples/Algorithms/Utilities/src/PrototracksToTracks.cpp

```diff
@@ -0,0 +1,91 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/Utilities/ProtoTracksToTracks.hpp"
+
+#include "Acts/Surfaces/PerigeeSurface.hpp"
+#include "ActsExamples/EventData/IndexSourceLink.hpp"
+#include "ActsExamples/EventData/MeasurementCalibration.hpp"
+#include "ActsExamples/EventData/ProtoTrack.hpp"
+#include "ActsExamples/EventData/SimSeed.hpp"
+#include "ActsExamples/Framework/WhiteBoard.hpp"
+#include "ActsExamples/Utilities/EventDataTransforms.hpp"
+
+#include <algorithm>
+
+namespace ActsExamples {
+
+PrototracksToTracks::PrototracksToTracks(Config cfg, Acts::Logging::Level lvl)
+    : IAlgorithm("PrototracksToTracks", lvl), m_cfg(std::move(cfg)) {
+  m_outputTracks.initialize(m_cfg.outputTracks);
+  m_inputProtoTracks.initialize(m_cfg.inputProtoTracks);
+  m_inputMeasurements.initialize(m_cfg.inputMeasurements);
+}
+
+ProcessCode PrototracksToTracks::execute(const AlgorithmContext& ctx) const {
+  auto trackContainer = std::make_shared<Acts::VectorTrackContainer>();
+  auto mtj = std::make_shared<Acts::VectorMultiTrajectory>();
+  TrackContainer tracks(trackContainer, mtj);
+
+  PassThroughCalibrator calibratorImpl;
+  MeasurementCalibratorAdapter calibrator(calibratorImpl,
+                                          m_inputMeasurements(ctx));
+
+  auto refSurface = Acts::Surface::makeShared<Acts::PerigeeSurface>(
+      Acts::Vector3{0., 0., 0.});
+  const auto& prototracks = m_inputProtoTracks(ctx);
+  ACTS_DEBUG("Received " << prototracks.size() << " prototracks");
+
+  float avgSize = 0;
+  std::size_t minSize = std::numeric_limits<std::size_t>::max();
+  std::size_t maxSize = 0;
+
+  for (const auto& protoTrack : prototracks) {
+    if (protoTrack.empty()) {
+      continue;
+    }
+
+    avgSize += protoTrack.size();
+    minSize = std::min(minSize, protoTrack.size());
+    maxSize = std::max(maxSize, protoTrack.size());
+
+    auto track = tracks.makeTrack();
+    for (auto idx : protoTrack) {
+      auto trackStateProxy =
+          track.appendTrackState(Acts::TrackStatePropMask::Calibrated);
+
+      IndexSourceLink sl(Acts::GeometryIdentifier{}, idx);
```

Comment:
Sorry I didn't see this before. Shouldn't the source links already exist? The geometry identifier is also incorrect (0) but should probably be the actual id.

Reply:
Yeah so this was supposed to be the minimal implementation that make the CKF Performance Writer work.
Currently, no downstream process of interest requires the geometry identifier.

Reply:
Sure, but if we start writing silently corrupt information, we'll eventually have to figure this out again through debugging when things break. So I think we should not introduce this information in the first place.